opta/nice_subprocess.py

```diff
@@ -22,79 +24,133 @@ def nice_run(  # type: ignore # nosec
-    exception will be raised.
-
-    There is an optional argument "input", allowing you to
-    pass bytes or a string to the subprocess's stdin.  If you use this argument
-    you may not also use the Popen constructor's "stdin" argument, as
-    it will be used internally.
-
-    By default, all communication is in bytes, and therefore any "input" should
-    be bytes, and the stdout and stderr will be bytes. If in text mode, any
-    "input" should be a string, and stdout and stderr will be strings decoded
-    according to locale encoding, or by "encoding" if set. Text mode is
-    triggered by setting any of text, encoding, errors or universal_newlines.
-
-    The other arguments are the same as for the Popen constructor.
-    """
-    if input is not None:
-        if kwargs.get("stdin") is not None:
-            raise ValueError("stdin and input arguments may not both be used.")
-        kwargs["stdin"] = PIPE
-
-    if capture_output:
-        if kwargs.get("stdout") is not None or kwargs.get("stderr") is not None:
-            raise ValueError(
-                "stdout and stderr arguments may not be used " "with capture_output."
-            )
-        kwargs["stdout"] = PIPE
-        kwargs["stderr"] = PIPE
-
-    with Popen(*popenargs, **kwargs) as process:
-        try:
-            stdout, stderr = process.communicate(input, timeout=timeout)
-        except TimeoutExpired as exc:
-            process.send_signal(signal.SIGINT)
-            # Wait again, now that the child has received SIGINT, too.
-            process.wait(timeout=exit_timeout)
-            process.kill()
-            if _mswindows:
-                # Windows accumulates the output in a single blocking
-                # read() call run on child threads, with the timeout
-                # being done in a join() on those threads.  communicate()
-                # _after_ kill() is required to collect that and add it
-                # to the exception.
-                exc.stdout, exc.stderr = process.communicate()
-            else:
-                # POSIX _communicate already populated the output so
-                # far into the TimeoutExpired exception.
-                process.wait()
-            raise
-        except KeyboardInterrupt:
-            print("Received keyboard interrupt")
-            # Wait again, now that the child has received SIGINT, too.
-            process.wait(timeout=exit_timeout)
-            raise
-        except Exception:  # Including KeyboardInterrupt, communicate handled that.
-            process.kill()
-            # We don't call process.wait() as .__exit__ does that for us.
-            raise
-        retcode = process.poll()
-        if check and retcode:
-            raise CalledProcessError(retcode, process.args, output=stdout, stderr=stderr)
-    return CompletedProcess(process.args, retcode or 0, stdout, stderr)
+
+    try:
+        Path("/tmp/optainput.tmp").touch()
```

Comment:
Use [`tempfile.NamedTemporaryFile`](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile) instead of trying to manage temporary files ourselves. Using the files in this way has at least two bugs:
- The file name is not random, so it may conflict with other instances of opta.
- If an exception is thrown, the file is not cleaned up.